pkg/operator/ceph/validation/validate.go

```diff
@@ -0,0 +1,96 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	rookclient "github.com/rook/rook/pkg/client/clientset/versioned"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"strconv"
+)
+
+var (
+	cephclusterResource   = metav1.GroupVersionResource{Group: "ceph.rook.io", Version: "v1", Resource: "cephclusters"}
+	rookClientSet         rookclient.Interface
+	universalDeserializer = serializer.NewCodecFactory(runtime.NewScheme()).UniversalDeserializer()
+)
+
+func SetRookClientSet(client rookclient.Interface) {
+	rookClientSet = client
+}
+func validateUpdatedCephCluster(updatedCephCluster cephv1.CephCluster, found *cephv1.CephCluster) error {
+	if isDataDirHostPathChanged(updatedCephCluster, found) {
+		return errors.New(fmt.Sprintf("invalid update : DataDirHostPath change from %q to %q is not allowed", found.Spec.DataDirHostPath, updatedCephCluster.Spec.DataDirHostPath))
+	}
+
+	if isHostNetworkChanged(updatedCephCluster, found) {
+		return errors.New(fmt.Sprintf("invalid update : HostNetwork change from %q to %q is not allowed", strconv.FormatBool(found.Spec.Network.HostNetwork), strconv.FormatBool(updatedCephCluster.Spec.Network.HostNetwork)))
+	}
+	return nil
+}
+
+func isHostNetworkChanged(updatedCephCluster cephv1.CephCluster, found *cephv1.CephCluster) bool {
+	return updatedCephCluster.Spec.Network.HostNetwork != found.Spec.Network.HostNetwork
+}
+func isDataDirHostPathChanged(updatedCephCluster cephv1.CephCluster, found *cephv1.CephCluster) bool {
+	return updatedCephCluster.Spec.DataDirHostPath != found.Spec.DataDirHostPath
+}
+
+func ValidateCephResource(request *v1beta1.AdmissionRequest) error {
+
+	// Checks if CephCluster is being updated and rejects the request if the dataDirHostPath is changed from initial value
+	if isCephClusterUpdate(request) {
+
+		// Fetch the existing ceph cluster object
+		found, err := fetchExistingCephCluster(request)
+		if err != nil {
+			return err
+		}
+
+		raw := request.Object.Raw
+		updatedCephCluster := cephv1.CephCluster{}
+		// Deserializing the incoming CephCluster object
+		if _, _, err := universalDeserializer.Decode(raw, nil, &updatedCephCluster); err != nil {
+			return errors.Wrap(err, "failed to deserialize pod object.")
+		}
+
+		err = validateUpdatedCephCluster(updatedCephCluster, found)
+		if err != nil {
+			return err
+		}
+
+	}
+
```

Comment:
Another check that would be interesting to implement is if `external: true`. In that case, you can't specify anything in the mon, mgr, dashboard, monitoring, network, disruptionManagement, or storage section of the cluster CR.

Reply:
Working on it

Reply:
Added this validation